share.android.js

```diff
@@ -68,13 +68,44 @@ export default class Share extends Component {
     })
   }
 
+  setRead(read) {
+    return fetch(`${this.state.server}/${API_VERSION}/bookmarks/${this.state.article.id}`, {
+      method: 'POST',
+      headers: {
+        'Authorization': `JWT ${this.state.token}`,
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        read: (read ? moment().format() : null),
+      }),
+    })
+  }
+
+  deleteArticle() {
+    return fetch(`${this.state.server}/${API_VERSION}/bookmarks/${this.state.article.id}`, {
+      method: 'DELETE',
+      headers: {
+        'Authorization': `JWT ${this.state.token}`,
+      },
+    })
+  }
+
   handleActionPress = (action) => {
     if (action === 'Cancel' || action === 'Close') {
       ShareExtension.close()
     } else if (action === 'Save') {
       this.saveArticle().then(
         () => ShareExtension.close(),
         () => ShareExtension.close())
+    } else if (action === 'Read' || action === 'Unread') {
+      this.setRead(action === 'Read').then(
+        () => ShareExtension.close(),
+        () => ShareExtension.close())
+    } else if (action === 'Delete') {
+      this.deleteArticle().then(
+        () => ShareExtension.close(),
+        () => ShareExtension.close())
```

Comment:
What do you think about switch to `switch`?

Bonus point to abstract: `action.then(() => close(), () => close())` (pseudo)